src/devdocs2zim/generator.py

```diff
@@ -0,0 +1,281 @@
+    long_description_format: str
+    # Semicolon delimited list of tags to apply to the ZIM.
+    tags: str
+
+    @staticmethod
+    def add_flags(parser: argparse.ArgumentParser, defaults: "ZimConfig"):
+        """Adds flags to the given parser with the given defaults."""
+        parser.add_argument(
+            "--output",
+            help=f"Output folder for ZIMs. Default: {defaults.output_folder}",
+            default=defaults.output_folder,
+            dest="output_folder",
+        )
+
+        parser.add_argument(
+            "--creator",
+            help=f"Name of content creator. Default: {defaults.creator!r}",
+            default=defaults.creator,
+        )
+
+        parser.add_argument(
+            "--publisher",
+            help=f"Custom publisher name. Default: {defaults.publisher!r}",
+            default=defaults.publisher,
+        )
+
+        parser.add_argument(
+            "--name-format",
+            help="Custom name format for individual ZIMs. "
+            f"Default: {defaults.name_format!r}",
+            default=defaults.name_format,
+            metavar="FORMAT",
+        )
+
+        parser.add_argument(
+            "--title-format",
+            help=f"Custom title format for individual ZIMs. "
+            f"Value will be truncated to {RECOMMENDED_MAX_TITLE_LENGTH} chars. "
+            f"Default: {defaults.title_format!r}",
+            default=defaults.title_format,
+            metavar="FORMAT",
+        )
+
+        parser.add_argument(
+            "--description-format",
+            help="Custom description format for individual ZIMs. Value will be "
+            f"truncated to {MAXIMUM_DESCRIPTION_METADATA_LENGTH} chars. "
+            f"Default: {defaults.title_format!r}",
+            default=defaults.description_format,
+            metavar="FORMAT",
+        )
+
+        parser.add_argument(
+            "--long-description-format",
+            help="Custom long description foramt for your ZIM. Value will be "
+            f"truncated to {MAXIMUM_LONG_DESCRIPTION_METADATA_LENGTH} chars. "
+            f"Default: {defaults.long_description_format!r}",
+            default=defaults.long_description_format,
+            metavar="FORMAT",
+        )
+
+        # Due to https://github.com/python/cpython/issues/60603 defaulting an array in
+        # argparse doesn't work so we expose the underlying semicolon delimited string.
+        parser.add_argument(
+            "--tags",
+            help="A semicolon (;) delimited list of tags to add to the ZIM."
+            "Formatting is supported. "
+            f"Default: {defaults.tags!r}",
+            default=defaults.tags,
+        )
+
+    @staticmethod
+    def of(namespace: argparse.Namespace) -> "ZimConfig":
+        """Parses a namespace to create a new ZimConfig."""
+        return ZimConfig.model_validate(namespace, from_attributes=True)
+
+    def format(self, placeholders: dict[str, str]) -> "ZimConfig":
```

Comment:
This is not yet used, correct?

Reply:
Correct, the next PR will run format on all the selected inputs to build the ZIM